src/dfmt/ast_info.d

```diff
@@ -448,6 +452,51 @@ final class FormatVisitor : ASTVisitor
         ternaryExpression.accept(this);
     }
 
+    override void visit(const FunctionCallExpression functionCall)
+    {
+        // Check if function has any arguments.
+        if (functionCall.arguments.namedArgumentList is null)
+        {
+            functionCall.accept(this);
+            return;
+        }
+
+        /+
+        Items are function arguments: f(<item>, <item>);
+        Iterate them and check if they are named arguments: tok!":" belongs to a
+        named argument if it is preceeded by one tok!"identifier" (+ any number
+        of comments):
+        +/
+        foreach (item; functionCall.arguments.namedArgumentList.items)
+        {
+            // Set to true after first tok!"identifier".
```

Comment:
this has a bug with e.g. `foo([a: b])` inserting that colon, since it just goes over the tokens. Make sure you check if the named argument is even a named argument and not a regular argument, since the grammar for named argument list is:

```
     * $(GRAMMAR $(RULEDEF namedArgument):
     *     ($(RULE identifer) $(LITERAL ':'))? $(RULE assignExpression)
     *     ;)
```

so check for `if (item.identifier != tok!"")` before looking for a colon

(I just realized the picked name is not great, but this was just added since the AST closely resembles the grammar which was changed like this in the spec)

Reply:
Thanks for the quick feedback, I will change the identifier check to `if (item.identifier != tok!"")`. The aa example works correctly (I think all nested expressions should), but using the information from libdparse is ofc better.

I didn't know `item.identifier` exists, where can I look these things up? So far I went with the files that code-d finds when using "go to definition". The libdparse that I get with dub when building has this class:
```d
final class NamedArgument : BaseNode
{
    override void accept(ASTVisitor visitor) const
    {
        mixin (visitIfNotNull!(name, assignExpression));
    }
    mixin OpEquals;
    /** */ Token name;
    /** */ ExpressionNode assignExpression;
    /** */ size_t startLocation;
    /** */ size_t endLocation;
}
```
I did find this http://libdparse.dlang.io/grammar.html#namedArgument , where `identifier` exists but not for example `startLocation`, can I expect that members listed on the website exist additionally to what I can see in the class definition?

Reply:
I wrote too soon, `item.identifier` isn't found, maybe I am using it wrong?
```d
foreach (item; functionCall.arguments.namedArgumentList.items)
{
    if (item.identifier == tok!"")
    {
        continue;
    }
    // ...
```
produces
```
src/dfmt/ast_info.d(472,21): Error: no property `identifier` for `item` of type `const(dparse.ast.NamedArgument)`
```

Reply:
ah sorry, it's `.name`

Yeah the ast.d file contains all the things, and to really understand where they come from in the grammar, you look at parser.d or the auto-generated docs on https://libdparse.dlang.io/grammar.html as well as possibly also the implementation